src/control/ros/Conversions.cpp

```diff
@@ -459,7 +460,16 @@ std::unique_ptr<SplineTrajectory> toSplineJointTrajectory(
   std::vector<aikido::statespace::ConstStateSpacePtr> subspaces;
   for (std::size_t citer = 0; citer < space->getDimension(); ++citer)
   {
-    subspaces.emplace_back(std::make_shared<aikido::statespace::R1>());
+    auto jointSpace = space->getJointSpace(citer);
+    auto r1Joint = std::dynamic_pointer_cast<const R1Joint>(jointSpace);
+    if (!r1Joint)
+    {
+      subspaces.emplace_back(std::make_shared<aikido::statespace::SO2>());
+    }
+    else
+    {
+      subspaces.emplace_back(std::make_shared<aikido::statespace::R1>());
+    }
```

Comment:
Why do we need to construct another space when we have the metaskeleton statespace which is R + SO(2)?

Reply:
@aditya-vk @brianhou Since the Metaskeleton statespaces are JointSpaces, I believe, they need to be converted / cast. Is there a better way to do this?